src/Tools/DynamoInstallDetective/Utilities.cs

```diff
@@ -55,16 +55,20 @@ public static IEnumerable LocateDynamoInstallations(string additionalDynamoPath,
         /// installations, we can use Dynamo as product search pattern and
         /// DynamoCore.dll as file search pattern.
         /// </summary>
-        /// <param name="productSearchPattern">search key for product</param>
+        /// <param name="productSearchPattern">search keys for product</param>
         /// <param name="fileSearchPattern">search key for files</param>
         /// <returns>List of KeyValuePair of product install location and 
         /// version info as Tuple of the file found in the installation based 
         /// on file search pattern. The returned list is sorted based on version 
         /// info.</returns>
-        public static IEnumerable FindProductInstallations(string productSearchPattern, string fileSearchPattern)
+        public static IEnumerable FindProductInstallations(List<string> productSearchPatterns, string fileSearchPattern)
```

Comment:
can't change this signature - it's public, you'll have to add another method and mark this obsolete.

Reply:
Will mark this obsolete and call the new version inside